include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -2395,5 +2395,22 @@ def ClassDeclOp
   let hasVerifier = 1;
 }
 
+def ClassNewOp
+    : MooreOp<
+          "class.new", [DeclareOpInterfaceMethods<MemoryEffectsOpInterface>]> {
+  let summary = "Allocate a new class instance";
+  let description = [{
+    Allocates a new instance of class @C. This op does not call the constructor.
```

Comment:
I'm guessing from the verifier that this is meant to eventually call the constructor once that's supported? If so we can probably drop the 'This op does not call the constructor.' since there's an unsupported error that gets thrown anyway (and it saves on some documentation that might get missed and end up outdated)

Reply:
Ah cool, sounds sensible!